### PR TITLE
pubkeytoaddr should return checksum'd address

### PR DESCRIPTION
**Pull request by Krigpl**
`IncomesKeeper` doesn't work well without it.
Changing `BLOCK_NUMBER_DB_KEY` will trigger listening to old events which will correctly mark awaiting incomes as confirmed.

### CODE REVIEW

File: golem/transactions/ethereum/ethereumincomeskeeper.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 
 class EthereumIncomesKeeper(IncomesKeeper):
-    BLOCK_NUMBER_DB_KEY = 'eth_incomes_keeper_block_number'
+    BLOCK_NUMBER_DB_KEY = 'eth_incomes_keeper_block_number_v2'
```
Comment (maaktweluit) [inline golem/transactions/ethereum/ethereumincomeskeeper.py:14]: How is this related to the checksum in the public address?

**Review by maaktweluit — APPROVED** [review]
LGTM!

One question but that is more that i do not know :)

File: golem/transactions/ethereum/ethereumincomeskeeper.py (hunk 1 above)
Comment (Krigpl) [reply golem/transactions/ethereum/ethereumincomeskeeper.py:14]: Because `pubkeytoaddr` didn't return checksum'd address IncomesKeeper didn't match any incomes correctly from blockchain events (which contain checksum'd addresses). Thus many events have been processed without correctly marking incomes as confirmed. Changing this value is one way to enforce listening to all income events from the beginning of the blockchain which in turn will correctly mark the incomes as confirmed.

**Review by mfranciszkiewicz — APPROVED** [review]

File: golem/transactions/ethereum/ethereumincomeskeeper.py (hunk 1 above)
Comment (maaktweluit) [reply golem/transactions/ethereum/ethereumincomeskeeper.py:14]: Super clear! thank you for the explanation!
